### PR TITLE
Add Collection view

### LINKED ISSUE #981: Add Collection view
Would be perfect to have a collection view only. 
You can recycle stuff from series, URL could be https://kodadot.xyz/rmrk/collections

Have cards of the collection showing pictures of the collection
Showing short stats like 
- number of items collected / total assets, 
- unique owners
- volume
- floor price
Adding pagination like showing only 3x3 would be enough 

Inspiration 
- https://singular.rmrk.app/collectibles?page=1&curated=true&verified=true&showCollections=yes
- https://solsea.io/collections

### PR DESCRIPTION
**Pull request by TK-Development**
**Thank you for your contribution** to the KodaDot NFT gallery.
Let's do a quick check before the merge.
Your contribution will be an eternal part of the community codebase.

### PR type
- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### Before submitting this PR, please make sure:
- [x] Code builds clean without any errors or warnings
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've didn't break any original functionality
- [x] I've posted screenshot of demonstrated change in this PR

### Optional
- [x] I've tested it on mobile and everything works
- [ ] I found edge cases

### What's new? (may be part of changelog)
- This PR resolves #981 - Add Collection View

### If issue had bounty label ($,$$,$$$)
- Fill up your KSM address: https://kodadot.xyz/transfer/?target=Ef14jyREXDkhsZV6wVfRqzgE2ayUe7fiK357ma6siwEpHtr

### Community participation
- [x] Are you at KodaDot Discord? If not yet, [let's join us - https://discord.gg/35hzy2dXXh](https://discord.gg/35hzy2dXXh)

### Screenshot
- [x] Whenever my fix has changed **something** on UI, a screenshot for others, is more than welcome

### CODE REVIEW

**Review by roiLeo — CHANGES_REQUESTED** [review]
Before diving into the code, let me suggest some things I would add or change.
- page title: something like "Explore all collections"
- n° of collection (totalCount)
- display collection issuer with Identity component
- hide collection without nfts?
- not a big fan of collection activity on Detail, instead I would have put collection short description
- maybe we should make a difference in design between a collection card and a gallery card
- searchbar ?

File: src/components/rmrk/Gallery/CollectionDetail.vue [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      <div class="level-item has-text-centered">
+        <div>
+          <p class="collection-title">{{ name }}</p>
+        </div>
+      </div>
+    </div>
+    <div class="level m-1" v-if="nfts">
+      <div class="level-item column is-6">
+        <div>
+          <p class="heading--inline heading">Items : <span class="money money--inline"> {{ collectionLength }} </span></p>
```
Comment (vikiival) [inline src/components/rmrk/Gallery/CollectionDetail.vue:13]: missing translations

File: src/components/rmrk/Gallery/Gallery.vue [hunk 2]
```diff
@@ -156,7 +155,7 @@ export default class Gallery extends Vue {
     sortBy: 'BLOCK_NUMBER_DESC',
     listed: false,
   };
-  private first = 12;
+  private first = 21;
```
Comment (vikiival) [inline src/components/rmrk/Gallery/Gallery.vue:158]: keep it between 9 - 12

File: src/queries/collectionListWithSearch.graphql [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    first: $first
+    offset: $offset
+    filter: {
+      id: { notLike: "%KAN%" }
+    }
+  ) {
+    totalCount
+    nodes {
+      ...collection
+      nfts(
```
Comment (vikiival) [inline src/queries/collectionListWithSearch.graphql:15]: Do you need NFTs here?

File: src/queries/collectionListWithSearch.graphql [hunk 4]
```diff
@@ -0,0 +1,32 @@
+#import "./collection.graphql"
+
+query collectionListWithSearch($first: Int!, $offset: Int) {
```
Comment (vikiival) [inline src/queries/collectionListWithSearch.graphql:3]: also I do not see any search here

File: src/components/rmrk/Gallery/CollectionDetail.vue [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+@Component({ components })
+export default class extends Vue {
+  @Prop() public nfts!: NFT[];
+  @Prop() public name!: string;
+
+  public yesterdayDate: Date = subDays(Date.now(), 1);
+
+  get saleEvents(): Interaction[] {
+    return this.nfts
```
Comment (vikiival) [inline src/components/rmrk/Gallery/CollectionDetail.vue:55]: move logic to the utils

File: src/components/rmrk/Gallery/Collections.vue [hunk 6]
```diff
@@ -0,0 +1,428 @@
+<template>
+  <div class="collections container">
+    <Loader :value="isLoading" />
+    <!-- TODO: Make it work with graphql -->
+    <b-field class="column">
+      <Pagination simple :total="total" v-model="currentValue" :perPage="perPage" replace class="is-right" />
+    </b-field>
+
+    <div>
+      <div class="columns is-multiline">
```
Comment (vikiival) [inline src/components/rmrk/Gallery/Collections.vue:10]: we have `GalleryCard`

**Review by vikiival — APPROVED** [review]

**Review by vikiival — CHANGES_REQUESTED** [review]
Nice work but: 
- missing component structure
- translations
- searchbar
- query takes too much unrelated info

File: src/components/rmrk/Gallery/CollectionDetail.vue (hunk 1 above)
Comment (TK-Development) [reply src/components/rmrk/Gallery/CollectionDetail.vue:13]: Got it. I'll update it.

File: src/components/rmrk/Gallery/Gallery.vue (hunk 2 above)
Comment (TK-Development) [reply src/components/rmrk/Gallery/Gallery.vue:158]: Got it. Set 12.

File: src/queries/collectionListWithSearch.graphql (hunk 3 above)
Comment (TK-Development) [reply src/queries/collectionListWithSearch.graphql:15]: Yes. To Display Item Count, Owned, etc.

File: src/components/rmrk/Gallery/Collections.vue (hunk 6 above)
Comment (TK-Development) [reply src/components/rmrk/Gallery/Collections.vue:10]: GalleryCard is using only for NFT, not Collection.

**Comment by TK-Development** [comment]
Before diving into the code, let me suggest some things I would add or change.

page title: something like "Explore all collections" - Yes, I'll change it.
n° of collection (totalCount) - You can see totalCount of collections in above pagination.
display collection issuer with Identity component - Where?
hide collection without nfts? - Shall I do this one?
not a big fan of collection activity on Detail, instead I would have put collection short description - The Issue 981 asked to display those activity.
maybe we should make a difference in design between a collection card and a gallery card - Shall I change the design of collections?
searchbar ? - What can I search in SearchBar?

**Comment by vikiival** [comment]
> searchbar ? - What can I search in SearchBar?

search by name or adress

**Comment by yangwao** [comment]
Hey let's add to navbar "COLLECTIONS"
image

and add it to landing page as well 
image

For detail per collection, 
- Make heading for collection much bigger
- spacing between second and third row I would do much less space.
- align it to the center

image

**Comment by yangwao** [comment]
I think maybe figure out mobile and we can merge it :)

Screenshot_20211104-192006

**Comment by TK-Development** [comment]
Changed Mobile Layout.

**Comment by yangwao** [comment]
Probably put 80% width for a header in name of the collection and for now, I guess we are good to go! 
There is still some cosmetic stuff we can tune later with the design.
Working Functionality first, design last 😄

image
image

**Comment by TK-Development** [comment]
Changed the width of title.
image
After finished this issue, I'll take a look at this issue.
Show only collection has NFT with various properties #1023

**Comment by yangwao** [comment]
@TK-Development I've sent you to pay out! https://kusama.subscan.io/extrinsic/0xe4e984cf692f06a76319756e0e6a9589c0bfaa8420868e42118991d60650e19b

**Comment by TK-Development** [comment]
Got it. Thanks!

**Pull request merged**